tests/components/file/test_config_flow.py

```diff
@@ -140,3 +155,49 @@ async def test_not_allowed(
 
     assert result2["type"] is FlowResultType.FORM
     assert result2["errors"] == {"file_path": "not_allowed"}
+
+
+@pytest.mark.parametrize(
+    ("platform", "data", "options", "new_options"),
+    [
+        (
+            "sensor",
+            MOCK_CONFIG_SENSOR,
+            MOCK_OPTIONS_SENSOR,
+            {CONF_UNIT_OF_MEASUREMENT: "mm"},
+        ),
+        ("notify", MOCK_CONFIG_NOTIFY, MOCK_OPTIONS_NOTIFY, {"timestamp": False}),
+    ],
+)
+async def test_options_flow(
+    hass: HomeAssistant,
+    mock_is_allowed_path: bool,
+    platform: str,
+    data: dict[str, Any],
+    options: dict[str, Any],
+    new_options: dict[str, Any],
+) -> None:
+    """Test options config flow."""
+    entry = MockConfigEntry(domain=DOMAIN, data=data, options=options, version=2)
+    entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "init"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input=new_options,
+    )
+    await hass.async_block_till_done()
```

Comment:
```suggestion
```

Reply:
This is needed for the state check a bit below (otherwise it might still be in progress)

Reply:
In my experience that isn't needed as there isn't anything in progress, because we're not sending events or anything like that

Reply:
There is something in progress as this particular test actually sets up the integration (I'm not mocking the setup here).
Depending on the speed this test would be quite flaky.